{{cookiecutter.github_repository}}/docs/backend/docker_setup.md

```diff
@@ -0,0 +1,74 @@
+# Getting Up and Running Locally With Docker
+
+The steps below will get you up and running with a local development environment. All of these commands assume you are in the root of your generated project.
+
+## Prerequisites
+
+- `Docker`; if you don’t have it yet, follow the [installation instructions](https://docs.docker.com/install/#supported-platforms).
+- `Docker Compose`; refer to the official documentation for the [installation guide](https://docs.docker.com/compose/install/).
+
+## Build the Stack
+
+This can take a while, especially the first time you run this particular command on your development system:
+
+```bash
+$ docker-compose -f local.yml build
+```
+
+For local env, use `local.yml`, for production use `production.yml` and then execute specific commmands necessary.
+
+## Run the Stack
+
+This brings up both Django and PostgreSQL. The first time it is run it might take a while to get started, but subsequent runs will occur quickly.
+
+```bash
+$ docker-compose -f local.yml up
+```
+
+To run in a detached (background) mode, just:
+
+```bash
+$ docker-compose up -d
+```
+
+## Execute Management Commands
+
+As with any shell command that we wish to run in our container, this is done using the `docker-compose -f local.yml run --rm` command:
+
+```bash
+$ docker-compose -f local.yml run --rm django python manage.py migrate
+$ docker-compose -f local.yml run --rm django python manage.py createsuperuser
+```
+
+Here, django is the target service we are executing the commands against.
+
+## Debugging
+
+### ipdb
+
+If you are using the following within your code to debug:
+
+```bash
+import ipdb; ipdb.set_trace()
+```
+
+Then you may need to run the following for it to work as desired:
+
+```bash
+$ docker-compose -f local.yml run --rm --service-ports django
+```
+
+### docker
+
+The `container_name` from the yml file can be used to check on containers with docker commands, for example:
+
+```bash
+$ docker logs worker
+$ docker top worker
+```
+
+## Celery Flower
+
+[Flower](https://github.com/mher/flower) is a “real-time monitor and web admin for Celery distributed task queue”.
+
+By default, it’s enabled both in local and production environments (`local.yml` and `production.yml` Docker Compose configs, respectively) through a `flower` service. For added security, `flower` requires its clients to provide authentication credentials specified as the corresponding environments’ `.envs/.local/.django` and `.envs/.production/.django` `CELERY_FLOWER_USER` and `CELERY_FLOWER_PASSWORD` environment variables. Check out `localhost:5555` and see for yourself.
```

Comment:
Should we document the role of the maintenance scripts? and how do use them?

Reply:
Done.